Formula/b/batt.rb

```diff
@@ -17,27 +17,27 @@ class Batt < Formula
   depends_on :macos
 
   def install
-    # batt does not provide separate control for AllowNonRootAccess.
-    # Normally it's changed during service file installation,
-    # but brew service supersedes that, leaving a daemon refusing to
-    # talk to anything non-root. Changing the defaults here.
-    inreplace "conf.go", "AllowNonRootAccess:      false,",
-                         "AllowNonRootAccess:      true,"
-    # Limit config path to Homebrew prefix. The socket remains in /var/run
-    # to deter non-root invocation of daemon
-    inreplace ["conf.go", "README.md"], "/etc/batt.json", etc/"batt.json"
-    # Due to local changes version tag would show v0.2.1-dirty
-    system "make", "VERSION=v#{version}"
+    # Point to the correct path for the binary
     inreplace "hack/cc.chlc.batt.plist", "/path/to/batt", bin/"batt"
-    system "plutil", "-insert", "ProcessType", "-string", "Background",
-                     "--", "hack/cc.chlc.batt.plist"
+    # Limit config path to Homebrew prefix.
+    system "plutil", "-insert", "ProgramArguments",
+           "-string", "--config=#{File.join(etc, "batt.json")}", "-append",
+           "--", "hack/cc.chlc.batt.plist"
+    # Allow non-root access to the battery controller.
+    system "plutil", "-insert", "ProgramArguments",
+           "-string", "--always-allow-non-root-access", "-append",
+           "--", "hack/cc.chlc.batt.plist"
+    # Due to local changes version tag would show vx.x.x-dirty, override VERSION.
+    # GOTAGS is set to disable built-in install/uninstall commands when building for Homebrew.
+    system "make", "GOTAGS=brew", "VERSION=v#{version}"
     bin.install "bin/batt"
     prefix.install "hack/cc.chlc.batt.plist"
   end
 
   def caveats
     <<~EOS
-      The service must be running before most of batt's commands will work.
+      1. Be sure to disable macOS's built-in Optimized Battery Charging feature.
+      2. The batt service must be running before most of batt's commands will work.
```

Comment:
We should probably keep the `caveats` as-is -- this is starting to look more like repeating upstream documentation than anything specific to Homebrew.

Reply:
Sure